trie/database.go

```diff
@@ -24,18 +24,33 @@ import (
 
 // StateDB is a wrapper around a polkadb
 type StateDB struct {
-	Db     polkadb.Database
+	Db     *polkadb.StateDB
```

Comment:
I think `trie/StateDb` should be replaced by `polkaddb/StateDb`, they are essentially equivalent in intent afaik (@dutterbutter)

Reply:
The struct field `Db` defined here is pointing to `polkadb/StateDB`, I don't think it needs replacing as we still want the other members of struct included here and we don't want them included in the `polkadb/StateDB` struct.

I think we need to change the name of this struct but I don't think it needs to be replaced.

Reply:
@noot Dutter and I just discussed this a bit further. It seems unnecessary to have both instances of the `StateDB`. Ultimately there are a number of trade-offs to how we implement the DBs and their wrappers. We need to have a discussion when possible to establish a plan moving forward

Reply:
okay for sure, I also found this pretty confusing. it would be nice to consolidate the 2 types but would definitely need a bit of code rewrite in the trie package. should I leave it for now?

Reply:
I think its fine to leave, but we should avoid touching the code again until we have a plan to move forward.